platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -296,51 +293,43 @@ private void initialiseDrawingSurface(MapboxMapOptions options) {
       mapRenderer = new TextureViewMapRenderer(getContext(), textureView, options.getLocalIdeographFontFamily()) {
         @Override
         protected void onSurfaceCreated(GL10 gl, EGLConfig config) {
-          MapView.this.post(new Runnable() {
-            @Override
-            public void run() {
-              // Initialise only once
-              if (mapboxMap == null) {
-                initialiseMap();
-                mapboxMap.onStart();
-              }
-            }
-          });
-
+          initRenderSurface();
           super.onSurfaceCreated(gl, config);
         }
       };
+
       addView(textureView, 0);
     } else {
       GLSurfaceView glSurfaceView = (GLSurfaceView) findViewById(R.id.surfaceView);
       glSurfaceView.setZOrderMediaOverlay(mapboxMapOptions.getRenderSurfaceOnTop());
-
       mapRenderer = new GLSurfaceViewMapRenderer(getContext(), glSurfaceView, options.getLocalIdeographFontFamily()) {
         @Override
         public void onSurfaceCreated(GL10 gl, EGLConfig config) {
-          MapView.this.post(new Runnable() {
-            @Override
-            public void run() {
-              // Initialise only once
-              if (mapboxMap == null) {
-                initialiseMap();
-                mapboxMap.onStart();
-              }
-            }
-          });
-
+          initRenderSurface();
           super.onSurfaceCreated(gl, config);
         }
       };
 
       glSurfaceView.setVisibility(View.VISIBLE);
-
     }
 
     nativeMapView = new NativeMapView(this, mapRenderer);
     nativeMapView.resizeView(getMeasuredWidth(), getMeasuredHeight());
   }
 
+  private void initRenderSurface() {
```

Comment:
Small nit. But could we rename this to something like `onSurfaceCreated`? It doesn't really initialise the render surface.

Reply:
https://github.com/mapbox/mapbox-gl-native/pull/11023